systemverilog-plugin/UhdmAst.cc

```diff
@@ -502,11 +499,19 @@ static void convert_packed_unpacked_range(AST::AstNode *wire_node)
         return false;
     }();
     if (convert_node) {
+        // if not already converted
         if (wire_node->multirange_dimensions.empty()) {
-            packed_size = add_multirange_attribute(wire_node, packed_ranges);
-            unpacked_size = add_multirange_attribute(wire_node, unpacked_ranges);
-            size = packed_size * unpacked_size;
-            ranges.push_back(make_range(size - 1, 0));
+            size_t packed_size = add_multirange_attribute(wire_node, packed_ranges);
+            size_t unpacked_size = add_multirange_attribute(wire_node, unpacked_ranges);
+            if (packed_ranges.size() == 1 && unpacked_ranges.empty()) {
+                ranges.push_back(packed_ranges[0]->clone());
+            } else if (unpacked_ranges.size() == 1 && packed_ranges.empty()) {
+                ranges.push_back(unpacked_ranges[0]->clone());
```

Comment:
I guess this solution is intentionally limited to 1 range. If so, we should at least add TODO/FIXME comment here describing that it works only for cases with one range.
Or maybe a loop that asserts every range starts with 0, if there is more than one.

Example that doesn't work:
```systemverilog
module top (wire_12_21);
  output [1:2] wire_12_21[2:1];
  wire [1:2] wire_12_21[2:1];
  assign wire_12_21[1][1] = 1'b0;
  assign wire_12_21[2][2] = 1'b1;
endmodule
```

Reply:
I've added warning, as we have 1 usage of multirange wire that doesn't start from 0 in opentitan.